cmake/BLTOptions.cmake

```diff
@@ -61,6 +61,7 @@ option(ENABLE_MPI          "Enables MPI support" OFF)
 option(ENABLE_OPENMP       "Enables OpenMP compiler support" OFF)
 option(ENABLE_CUDA         "Enable CUDA support" OFF)
 
+option(ENABLE_GMOCK        "Enable Google Mock testing support" OFF)
```

Comment:
I think so.

It would be good to think about a clean way to set up sub-options for a given TPL/component (e.g. to include the name of the TPL/component that is is related to).